src/main/kotlin/creator/custom/types/LicenseProperty.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Minecraft Development for IntelliJ
+ *
+ * https://mcdev.io/
+ *
+ * Copyright (C) 2024 minecraft-dev
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, version 3.0 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.demonwav.mcdev.creator.custom.types
+
+import com.demonwav.mcdev.creator.custom.TemplatePropertyDescriptor
+import com.demonwav.mcdev.creator.custom.model.LicenseData
+import com.demonwav.mcdev.util.License
+import com.intellij.ide.util.projectWizard.WizardContext
+import com.intellij.openapi.observable.properties.GraphProperty
+import com.intellij.openapi.observable.properties.PropertyGraph
+import com.intellij.openapi.observable.util.transform
+import com.intellij.ui.ComboboxSpeedSearch
+import com.intellij.ui.EnumComboBoxModel
+import com.intellij.ui.dsl.builder.Panel
+import com.intellij.ui.dsl.builder.bindItem
+import java.time.ZonedDateTime
+
+class LicenseProperty(
+    graph: PropertyGraph,
+    descriptor: TemplatePropertyDescriptor,
+    properties: Map<String, CreatorProperty<*>>
+) : CreatorProperty<LicenseData>(descriptor, graph, properties) {
+
+    override val graphProperty: GraphProperty<LicenseData> =
+        graph.property(createDefaultValue(descriptor.default))
+
+    override fun createDefaultValue(raw: Any?): LicenseData =
+        deserialize(raw as? String ?: License.ALL_RIGHTS_RESERVED.id)
+
+    override fun serialize(value: LicenseData): String = value.id
+
+    override fun deserialize(string: String): LicenseData =
+        LicenseData(string, License.byId(string)?.toString() ?: string, ZonedDateTime.now().year.toString())
```

Comment:
Maybe licenses could be made extensible, doesn't have to be in this PR though.

Reply:
Yes, I simply moved the licenses to the templates repo and reused existing code.
The ids and corresponding files could be provided in a shared descriptor that other descriptors inherit from.